src/overlays/actors/ovl_En_Eiyer/z_en_eiyer.c

```diff
@@ -48,69 +74,636 @@ static ColliderCylinderInit D_80A01970 = {
+    this->actor.speedXZ *= 1.1f;
+
+    if (this->actor.bgCheckFlags & 8 || this->actor.bgCheckFlags & 1) {
+        EnEiyer_SetupLand(this);
+    }
+
+    if (this->colCyl.base.atFlags & AT_HIT) {
+        this->colCyl.base.atFlags &= ~(AT_ON | AT_HIT);
+    }
+
+    func_8002F974(&this->actor, NA_SE_EN_EIER_FLY - SFX_FLAG);
+}
+
+void EnEiyer_Land(EnEiyer *this, GlobalContext *globalCtx) {
+    SkelAnime_Update(&this->skelanime);
+    Math_ScaledStepToS(&this->actor.world.rot.x, -0x4000, 0x450);
+    Math_StepToF(&this->actor.speedXZ, 7.0f, 1.0f);
+
+    if (this->actionTimer == -1) {
+        if (this->actor.bgCheckFlags & 8 || this->actor.bgCheckFlags & 1) {
+            this->actionTimer = 0xA;
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x1E, NA_SE_EN_OCTAROCK_SINK);
+            if (this->actor.bgCheckFlags & 1) {
+                EffectSsGSplash_Spawn(globalCtx, &this->actor.world.pos, NULL, NULL, 1, 700);
+            }
+        }
+    } else {
+        if (this->actionTimer != 0) {
+            this->actionTimer--;
+        }
+        if (this->actionTimer == 0) {
+            this->actor.shape.rot.x = 0;
+            this->actor.world.rot.x = 0;
+            EnEiyer_SetupAppearFromGround(this);
+        }
+    }
+}
+
+void EnEiyer_Hurt(EnEiyer *this, GlobalContext *globalCtx) {
+    SkelAnime_Update(&this->skelanime);
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+
+    Math_ApproachF(&this->basePos.y, this->actor.floorHeight + 80.0f + 5.0f, 0.5f, this->actor.speedXZ);
+    this->actor.world.pos.y = this->basePos.y - 5.0f;
+
+    if (this->actor.bgCheckFlags & 8) {
+        this->targetYaw = this->actor.wallYaw;
+    } else {
+        this->targetYaw = this->actor.yawTowardsPlayer + 0x8000;
+    }
+
+    Math_ScaledStepToS(&this->actor.world.rot.y, this->targetYaw, 0x38E);
+    Math_ScaledStepToS(&this->actor.shape.rot.x, 0, 0x200);
+    this->actor.shape.rot.z = sinf(this->actionTimer * (M_PI/5)) * 5120.0f;
+
+    if (this->actionTimer == 0) {
+        this->actor.shape.rot.x = 0;
+        this->actor.shape.rot.z = 0;
+        this->colCyl.base.acFlags |= AC_ON;
+        EnEiyer_SetupGlide(this);
+    }
+    this->actor.world.rot.x = -this->actor.shape.rot.x;
+}
+
+void EnEiyer_Die(EnEiyer *this, GlobalContext *globalCtx) {
+    SkelAnime_Update(&this->skelanime);
+
+    if (this->actor.speedXZ > 0.0f) {
+        Math_ScaledStepToS(&this->actor.shape.rot.x, -0x4000, 0x400);
+    } else {
+        Math_ScaledStepToS(&this->actor.shape.rot.x, 0x4000, 0x400);
+    }
+
+    this->actor.shape.rot.z += 0x1000;
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+
+    this->actor.world.rot.x = -this->actor.shape.rot.x;
+    if (this->actionTimer == 0 || this->actor.bgCheckFlags & 0x10) {
+        EnEiyer_SetupDead(this);
+    }
+}
+
+void EnEiyer_Dead(EnEiyer *this, GlobalContext *globalCtx) {
+    if ((s16)(this->actor.shape.shadowAlpha - 5) < 0) {
```

Comment:
```suggestion
    if (this->actor.shape.shadowAlpha - 5 < 0) {
```
If possible.

Reply:
necessary to match :\

Reply:
This works, idk if it's better
```c
    this->actor.shape.shadowAlpha = MAX(0, (s16)(this->actor.shape.shadowAlpha - 5));
```

Reply:
If you can get it to work with a CLAMP macro that would be ideal.